client/src/telethon/_impl/client/events/filters/combinators.py

```diff
@@ -81,8 +82,10 @@ def filters(self) -> tuple[Filter, ...]:
         """
         return self._filters
 
-    def __call__(self, event: Event) -> bool:
-        return any(f(event) for f in self._filters)
+    async def __call__(self, event: Event) -> bool:
+        return any(
+            [await r if isawaitable(r := f(event)) else r for f in self._filters]
```

Comment:
This is no longer lazy. It evaluates the entire list upfront.

If removing the `[]` doesn't work, it should be rewritten as a `for` loop.